### PR TITLE
refactor: Remove deprecated bcrypt-nodejs package

### LINKED ISSUE #8903: bcrypt-nodejs is deprecated and throws a warning on install
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

Causes a warning during install, `npm WARN deprecated bcrypt-nodejs@0.0.3: bcrypt-nodejs is no longer actively maintained. Please use bcrypt or bcryptjs. See https://github.com/kelektiv/node.bcrypt.js/wiki/bcrypt-vs-brypt.js to learn more about these two options`

### Steps to reproduce

Install the latest version of the parse-server using `npm ci`

### Actual Outcome

Installs, but throws the warning in the console output.

### Expected Outcome

The warning not to be shown.

### Environment

Server
- Parse Server version: `6.4.0` and `alpha`
- Operating system: `linux`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): n/a
- Database version: n/a
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): n/a

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): n/a
- SDK version: n/a

### PR DESCRIPTION
**Pull request by cbaker6**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Causes a warning during install, `npm WARN deprecated bcrypt-nodejs@0.0.3: bcrypt-nodejs is no longer actively maintained. Please use bcrypt or bcryptjs. See https://github.com/kelektiv/node.bcrypt.js/wiki/bcrypt-vs-brypt.js to learn more about these two options`

Closes: #8903 

## Tasks

- [x] Ensure CI passes

### CODE REVIEW

File: spec/Auth.spec.js [hunk 1]
```diff
@@ -81,17 +81,6 @@ describe('Auth', () => {
         .then(roles => expect(roles).toEqual([]))
         .then(() => done());
     });
-
-    it('should properly handle bcrypt upgrade', done => {
```
Comment (cbaker6) [inline spec/Auth.spec.js:85]: This is an old test added 8+ years ago that was introduced in #2549. I don't believe it's needed anymore.

**Comment by cbaker6** [comment]
Ready for review...

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.19](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.19)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)

**Pull request merged**